docs/DeveloperGuide.md

```diff
@@ -1,38 +1,239 @@
+| v1.0    | - User who wants to start working out<br/>- User who wants to train a specific part of my body           | - Select a specific intensity workout<br/>- Request a workout that comprises exercises that thoroughly exercises that body part | - Exercise according to selected intensity<br/> - Work on my target body part. |
+| V1.0    | - User                                                                                                   | - be able to filter my exercises by body part                                                                                   | - train a specific part of my body                                             |
+| V2.0    | - User                                                                                                   | - be able to start a workout and set it as complete and save it                                                                 | - reflect and track my progress                                                |
+| V2.0    | - User looking for motivation                                                                            | - be able to track my workout history as statistics                                                                             | - better visualise my overall progress                                         |
+| V2.0    | - user with little to no experience with exercise                                                        | - be given instructions for the specific exercise that I am working on                                                          | be educated on how to complete the exercise correctly                          |
+| V2.0    | - User who wants to stay motivated to workout </br> - User who wants to feel good about my past workouts | - See myself be able to accomplish or achieve incrementally greater goals </br> - Keep track of all my exercises                | - Continue to stay motivated in making exercise a fun, long-lasting habit      |
+
+## Non-Functional Requirements
+1. The program should be able to generate a list of exercises within 5 seconds
+2. The program should be able to run on any PC (i.e. different OS) <br/> - Be able to handle cases such as user data file corruption
+3. The program is not required to ensure the workouts are carried out safely and properly by the user
+4. Avoid giving repeated workouts in the same session
+
+## Glossary
+
+* *FitnessDuke* - The name of our application
+* *OS* - Operating Systems - Linux, MacOS, Windows
+* *CLI* - Command Line Interface - The terminal in the PC OS
+* *ExerciseData* - Individual Exercise data from the ```data.json``` file
+* *Ui* - User Interface
+
+## Instructions for manual testing
+
+{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+
+### Launch and shutdown of program

```

Comment:
The launch of the program can be more comprehensive, such as which version of Java, etc.